### ISSUE TITLE
Review: MoreEnumerable.Rank() and MoreEnumerable.RankBy()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [blyzer/morelinq#55]
```
Purpose of code changes on this branch:

Rev 149: Implementation of Rank returns a sequences of positive numbers
representing the rank of items in the source sequence, with the lowest
values corresponding to the "highest" value in the sequence. RankBy works
similarly by allows ranking to be based on a key determined by a caller
supplied function. Overload also exist that accept an IComparer<T> allowing
the comparison semantics to be defined by the caller.

When reviewing my code changes, please focus on:

* The public interface of the extension method.
* The effectiveness and clarity of the available XML comment documentation.
* How well this operator fits into the MoreLINQ ecosystem of
extension methods.
* The extent of coverage available from the corresponding unit tests.
* Whether edge cases are correctly identified and handled in the
implementation and tests.

After the review, I'll merge this branch into:
/trunk

```

Original issue reported on code.google.com by `ambientl...@gmail.com` on 25 Jan 2010 at 12:10

**Comment by GoogleCodeExporter** [comment]
```
Now migrated to Hg and available in clone:
http://code.google.com/r/azizatif-morelinq-evenmore/
This clone is ready to be cloned :O) for further review. It also has the 
main/default branch already merged in to bring it up to date.
```

Original comment by `azizatif` on 25 May 2012 at 11:38

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 10 Jun 2012 at 3:35
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Following no response from ambientlion (who possibly/assuming lost interest 
given it has taken years to get this far), taking ownership of issue to 
consider as addition for MoreLINQ 2.0 milestone.
```

Original comment by `azizatif` on 12 Jun 2013 at 10:10
- Added labels: **Component-Logic**, **Milestone-Release2.0**, **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Review**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ecbb044b3f86.
```

Original comment by `azizatif` on 14 Jun 2013 at 5:18
- Changed state: **Fixed**
